projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -115,7 +116,13 @@ def run_step(context: JobContext, step: Step) -> None:
         out: HookCallResult
         out = yield
 
-        exception = out.get_result().exception
+        result: StepResult
+        if out.excinfo:
+            result = context.core_context.state.get("step_results")[-1]
```

Comment:
I'm not sure if and where I should document this behaviour. It might be relevant for plugins that want to hook to `run_step` and get the step result. It's a bit convoluted.